tmt/steps/execute/__init__.py

```diff
@@ -595,19 +636,41 @@ def prepare_tests(self, guest: Guest, logger: tmt.log.Logger) -> list[TestInvoca
 
         return invocations
 
+    def _render_script_template(self, source: Path, context: dict[str, str]) -> Path:
```

Comment:
This looks like the implementation of a script class leaked into the plugin.

I'd suggest extending `Script` to become a context manager, i.e. adding `__enter__` and `__exit__` methods. The default implementation of `__enter__` would return `self.path`, but `ScriptTemplate` class would do the magic we have here, render the template and write down the path to temporary file. Then do `with script as path` below, and `ScriptTemplate`'s `__exit__` would remove the temporary file when leaving the context. Plugin does not need to be concerned about itnernals of various script implementations, all it needs is a local path to a file, and the rest shouldn't be its responsibility.

Reply:
ack. nice idea, working it out

Reply:
@happz should be resolved. First time I used context manager in the code I wrote, it really makes the code a lot nicer.